src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -663,7 +663,7 @@ private bool TryReadEvent(out NotifyEvent notifyEvent)
                 //         uint32_t len;
                 //         char     name[]; // length determined by len; at least 1 for required null termination
                 //     };
```

Comment:
I'm unable to comment on the line I want to, but line 621

 Debug.Assert(_bufferPos >= 0 && _bufferPos <= _bufferAvailable);

Should the test be "_bufferPos < bufferAvailable" or "_bufferPos < buffer.Length"?